site/src/data/derived.js

```diff
@@ -1,34 +1,129 @@
-import { uniq } from "lodash-es";
+import { sortBy, uniq } from "lodash-es";
 
 /** years old before not "new" anymore */
 export const newThreshold = 2;
 
-/** derive/compute some props from existing props on datum  */
-export const deriveDatum = (d) => {
+/** derive/compute some props from existing props on locus  */
+export const deriveLocus = (locus, loci, citations) => {
   /** keep existing data */
-  d = { ...d };
+  locus = { ...locus };
+
+  /** nice looking id, for labels */
+  locus.nice_id = locus.id.replace("_", " ");
 
   /** construct full position strings */
   for (const assembly of ["hg19", "hg38", "t2t"])
-    d[`position_${assembly}`] =
-      `${d.chrom}:${d[`start_${assembly}`]}-${d[`stop_${assembly}`]}`;
+    locus[`position_${assembly}`] =
+      `${locus.chrom}:${locus[`start_${assembly}`]}-${locus[`stop_${assembly}`]}`;
 
   /** make prevalence fraction */
-  if (d.prevalence)
-    d.prevalence = d.prevalence
+  if (locus.prevalence)
+    locus.prevalence = locus.prevalence
       .split("/")
       .map(Number)
       .map((value) => (value || 0).toLocaleString());
 
   /** init tags */
-  d.locus_tags ??= [];
+  locus.locus_tags ??= [];
 
   /** add tags */
-  if (new Date().getFullYear() - parseInt(d.year) <= newThreshold)
-    d.locus_tags.push("new");
+  if (new Date().getFullYear() - parseInt(locus.year) <= newThreshold)
+    locus.locus_tags.push("new");
 
   /** clean up tags */
-  d.locus_tags = uniq(d.locus_tags.filter(Boolean));
+  locus.locus_tags = uniq(locus.locus_tags.filter(Boolean));
+
+  if (loci) {
+    /** find other loci that are the same gene */
+    locus.genes = loci.filter(({ gene }) => gene === locus.gene);
+  }
+
+  if (citations) {
+    /** map for quick lookup of citation by id */
+    const citationLookup = Object.fromEntries(
+      citations.map((citation) => [citation.id, citation]),
+    );
+
+    /** fields that may contain in-text citations, in order they appear on locus page (so number increases as you go down page) */
+    const inTextFields = [
```

Comment:
Here is where we specify which fields, and in what order, to extract (and thus number) citations. Remember to update this if you re-order these fields on the page, or if you add a new field that has in-text references.